### DISCUSSION TITLE
Cant register Flask context to sqlalchemy instance

### DISCUSSION THREAD

**Discussion by CypherpunkSamurai** [pallets-eco/flask-sqlalchemy#1190] [Q&A]
Hello 👋🏼,

I'm working on a flask postgresql project that uses flask_sqlalchemy. When I run the flask app and the controller route get a get request it gives error, though I have init the db when creating the flask app.
```py
RuntimeError: The current Flask app is not registered with this 'SQLAlchemy' instance. Did you forget to call 'init_app', or did you create multiple 'SQLAlchemy' instances?
```

Sources Here: https://github.com/CypherpunkSamurai/flask-restapi/blob/master/app/run.py

I have the following file structure:
```
- app
  - posts\
    - models.py
  - utils.py
  - run.py
```

`run.py` has the following factory.
```py
# here we import the db from models
from posts.models import db
...
def create_app():
    """
    Flask App Factory to create app (recommended approach by flask)
    https://flask.palletsprojects.com/en/2.2.x/patterns/appfactories/

    :return:
    """
    app = Flask(__name__)

    # Blueprints
    app.register_blueprint(post_routes.b_new_post)
    app.register_blueprint(post_routes.b_get_post)
    app.register_blueprint(weather_routes.b_get_weather)

    # SQL Config
    app.config["SQLALCHEMY_DATABASE_URI"] = utils.POSTGRESQL_URI
    app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = utils.IS_DEBUG

    # Context
    with app.app_context():
        db.init_app(app)
    return app
```

`models.py` has:
```py
# use sqlalchemy
from flask_sqlalchemy import SQLAlchemy

db = SQLAlchemy()
```

**Comment by ThiefMaster** [comment]
Running queries at import time is a terrible idea. Did you forget to wrap your stuff in `controller.py` in a function so it only runs when a request is received?

**Reply by CypherpunkSamurai** [reply]
It could be 🤔

The only thing running is `db.create_all()` if the `DEBUG` var is `True`. I also moved the db from controller to models to keep it in one file.

I think what might be happening is the db.Column etc used for the models is initialized when the db has the empty app context.

can you suggest what might be wrong here looking at the updated code [here](https://github.com/CypherpunkSamurai/flask-restapi/blob/5bf096720787576a832b33610a4a11485c8a26e3/app/run.py#L51) and [here](https://github.com/CypherpunkSamurai/flask-restapi/blob/master/app/posts/models.py#L26)?

**Reply by ThiefMaster** [reply]
well looks like your code you posted here doesn't match what's actually in your repo...

please post the error with a full traceback. we don't want to look through your whole code to even see where the exception gets raised...

**Reply by CypherpunkSamurai** [reply]
sure. just a second.

**Reply by CypherpunkSamurai** [reply]
cool. updated it :)
